### PR TITLE
Sysext image for VMware OEM

### PR DESCRIPTION
**Pull request by krnowak**
CI: http://jenkins.infra.kinvolk.io:8080/job/container/job/packages/4451/cldsv/

- net-libs/rpcsvc-proto: [PROD] [DEV]
  - from 1.4.2 to 1.4.4
  - moved from overlay to portage-stable and added to automation
    - there were no Flatcar modifications in the package
  - EAPI 8
  - moved gettext dependency from DEPEND to BDEPEND
  - extended the preprocessor detection workaround
  - release notes: https://github.com/thkukuk/rpcsvc-proto/releases/tag/v1.4.4

- dev-libs/xmlsec: [VMWARE]
  - new package, at 1.3.1
  - a dependency of app-emulation/open-vm-tools

- dev-libs/libmspack: [VMWARE]
  - still at 0.10.1_alpha-r1
  - moved from overlay to portage-stable and added to automation
    - Flatcar modifications were dropped, as they were about putting stuff in to /oem
  - EAPI 8
  - dropped static-libs USE flag

- sys-fs/fuse-common: [VMWARE]
  - new package, at 3.10.4
  - a dependency of sys-fs/fuse

- sys-fs/fuse: [VMWARE]
  - new package, at 3.16.1
  - a dependency of app-emulation/vmware
  - pulling in an unstable version, as it fixes an issue with cross-compilation

- dev-libs/libdnet: [VMWARE]
  - from 1.14-r3 to 1.16.2
  - moved from overlay to portage-stable and added to automation
    - Flatcar modifications were dropped, as they were about putting stuff in to /oem and dropping support for python
  - EAPI 8
  - added dependency on dev-libs/libbsd
  - release notes: https://github.com/ofalk/libdnet/releases/tag/libdnet-1.16.2 https://github.com/ofalk/libdnet/releases/tag/libdnet-1.16

- app-emulation/open-vm-tools: [VMWARE]
  - from 12.3.0 to 12.3.0-r1
  - synced with Gentoo, dropping all the OEM modifications
  - all the modifications are planned to be upstreamed to Gentoo
  - enabled vmhgfs-fuse
  - enabled installation of udev rules
  - enabled vgauth
  - enabled pam
  - added a user patch to tell the configure script to use kernel version from sys-kernel/coreos-kernel
  - containerinfo is still disabled, will work on enabling it later

--

- [x] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)
- [x] Inspected CI output for image differences: `/boot` and `/usr` size, packages, list files for any missing binaries, kernel modules, config files, kernel modules, etc.

### CODE REVIEW

File: sdk_container/src/third_party/coreos-overlay/coreos-base/oem-vmware/files/open-vm-tools-sysext-upholds.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[Unit]
+Upholds=vmtoolsd.service vgauthd.service
```
[the remark this answers is not in the document]
Comment (krnowak) [reply sdk_container/src/third_party/coreos-overlay/coreos-base/oem-vmware/files/open-vm-tools-sysext-upholds.conf:2]: Yeah, it's new. `vmhgfs-fuse` is also new.

**Comment by krnowak** [comment]
CI passed.

File: changelog/changes/2023-09-20-vmware-sysext.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+- Rework VMware OEM to become a sysext image.
+- The open-vm-tools package in VMware OEM now comes with vmhgfs-fuse, udev rules, pam and vgauth.
```
Comment (pothos) [inline changelog/changes/2023-09-20-vmware-sysext.md:2]:
```suggestion
- Reworked the VMware OEM software to be shipped a A/B updated systemd-sysext image
- The open-vm-tools package in VMware OEM now comes with vmhgfs-fuse, udev rules, pam and vgauth
```

File: changelog/changes/2023-09-20-vmware-sysext.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+- Reworked the VMware OEM software to be shipped a A/B updated systemd-sysext image
```
Comment (pothos) [inline changelog/changes/2023-09-20-vmware-sysext.md:1]:
```suggestion
- Reworked the VMware OEM software to be shipped as A/B updated systemd-sysext image
```

**Review by pothos — APPROVED** [review]
Thanks a lot

File: changelog/changes/2023-09-20-vmware-sysext.md (hunk 3 above)
Comment (pothos) [reply changelog/changes/2023-09-20-vmware-sysext.md:1]: (that was my typo)

**Pull request merged**